torch/quantization/qconfig.py

```diff
@@ -122,7 +122,61 @@ def get_default_qat_qconfig(backend='fbgemm'):
         qconfig = default_qat_qconfig
     return qconfig
 
-def assert_valid_qconfig(qconfig: Optional[Union[QConfig, QConfigDynamic]],
+class QConfigWithModule(namedtuple('QConfigWithModule', ['activation', 'weight'])):
+    """
+
+    It's like QConfig and QConfigDynamic but intended to be used once the qconfig is placed
+    onto a specific module. if create_qconfig_on_module is used, this includes transforming the activation 
+    and weight obs constructors such that they create the obs on the same device as module.
+
+    The __new__ method to create this type of object is not intended to be used directly, 
+    instead use create_qconfig_with_module(qconfig,module). Without __new__ set up like this, deepcopy
+    will not work.
+
+    Note that QConfigWithModule needs to contain observer **classes** (like MinMaxObserver) or a callable constructor that returns
+    instances on invocation, not the concrete observer instances themselves (like MinMaxObserver()).
+    Quantization function will instantiate observers multiple times for each of the layers.
+
+    Observer classes have usually reasonable default arguments, but they can be overwritten with `with_args`
+    method (that behaves like functools.partial). The observer classes/constructors in QConfigWithModule are intended 
+    to be used with `with_callable_args` which operates similarly to `with_args` but calls the arg when the constructor is invoked.
+    This is how the constructors check the module device at invocation time rather than when the constructor is set up.
+    """
+    def __new__(cls, activation=torch.nn.Identity, weight=torch.nn.Identity):
+        # catch common mistakes
+        if isinstance(weight, nn.Module):
+            raise ValueError("QConfigWithModule received observer instance, please pass observer class instead. " +
+                             "Use MyObserver.with_args(x=1) to override arguments to constructor if needed")
+        return super(QConfigWithModule, cls).__new__(cls, activation, weight)
+
+
+def create_qconfig_with_module(
```

Comment:
could we check if this is going to survive serialization?  For example, at any time the user should be able to save their model, load it from disk, and everything should be the same.

Reply:
TestPostTrainingDynamic.test_quantized_rnn and python test/test_quantization.py TestObserver.test_per_channel_observers both pass, (as do TestPostTrainingDynamic and TestObserver).

Reply:
I also added a new test case, seems like no issue though it looks like qconfig isn't serialized, and since already created observers were mostly functioning correctly, I don't think this change affects serialization as far as I can tell.